iodata/test/test_fchk.py

```diff
@@ -514,3 +515,184 @@ def test_load_nbasis_indep(tmpdir):
+
+
+def test_dump_fchk_from_fchk_he(tmpdir):
+    check_load_dump_consistency(tmpdir, 'he_s_orbital.fchk')
+    check_load_dump_consistency(tmpdir, 'he_sp_orbital.fchk')
+    check_load_dump_consistency(tmpdir, 'he_spd_orbital.fchk')
+    check_load_dump_consistency(tmpdir, 'he_spdf_orbital.fchk')
+    check_load_dump_consistency(tmpdir, 'he_spdfgh_orbital.fchk')
+    check_load_dump_consistency(tmpdir, 'he_s_virtual.fchk')
+    check_load_dump_consistency(tmpdir, 'he_spdfgh_virtual.fchk')
+
+
+def test_dump_fchk_from_fchk_o2(tmpdir):
+    check_load_dump_consistency(tmpdir, 'o2_cc_pvtz_cart.fchk')
+    check_load_dump_consistency(tmpdir, 'o2_cc_pvtz_pure.fchk')
+
+
+def test_dump_fchk_from_fchk_water_dimer_ghost(tmpdir):
+    check_load_dump_consistency(tmpdir, 'water_dimer_ghost.fchk')
+
+
+def test_dump_fchk_from_molden_f(tmpdir):
+    check_load_dump_consistency(tmpdir, 'F.molden')
+
+
+def test_dump_fchk_from_molden_ne(tmpdir):
+    check_load_dump_consistency(tmpdir, 'neon_turbomole_def2-qzvp.molden')
+
+
+def test_dump_fchk_from_molden_he2(tmpdir):
+    check_load_dump_consistency(tmpdir, 'he2_ghost_psi4_1.0.molden')
+
+
+def test_dump_fchk_from_molden_nh3(tmpdir):
+    check_load_dump_consistency(tmpdir, 'nh3_orca.molden')
+    check_load_dump_consistency(tmpdir, 'nh3_psi4.molden')
+    check_load_dump_consistency(tmpdir, 'nh3_psi4_1.0.molden')
+    check_load_dump_consistency(tmpdir, 'nh3_molpro2012.molden')
+    check_load_dump_consistency(tmpdir, 'nh3_molden_cart.molden')
+    check_load_dump_consistency(tmpdir, 'nh3_molden_pure.molden')
+    check_load_dump_consistency(tmpdir, 'nh3_turbomole.molden')
+
+
+def test_dump_fchk_from_wfn_he(tmpdir):
+    check_load_dump_consistency(tmpdir, 'he_s_virtual.wfn')
+    check_load_dump_consistency(tmpdir, 'he_s_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_p_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_d_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_sp_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_spd_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_spdf_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_spdfgh_orbital.wfn')
+    check_load_dump_consistency(tmpdir, 'he_spdfgh_virtual.wfn')
+
+
+def test_dump_fchk_from_wfn_li(tmpdir):
+    check_load_dump_consistency(tmpdir, 'li_sp_virtual.wfn')
+    check_load_dump_consistency(tmpdir, 'li_sp_orbital.wfn')
+
+
+def test_dump_fchk_from_wfn_lih_cation(tmpdir):
+    check_load_dump_consistency(tmpdir, 'lih_cation_uhf.wfn')
+    check_load_dump_consistency(tmpdir, 'lih_cation_rohf.wfn')
+
+
+def test_dump_fchk_from_wfn_cisd_lih_cation(tmpdir):
+    check_load_dump_consistency(tmpdir, 'lih_cation_cisd.wfn')
+
+
+# def test_dump_fchk_from_wfn_fci_lih_cation(tmpdir):
+#     check_load_dump_consistency(tmpdir, 'lih_cation_fci.wfn')
+
+
+# def test_dump_fchk_from_wfn_fci_lif(tmpdir):
+#     check_load_dump_consistency(tmpdir, 'lif_fci.wfn')
+
+
```

Comment:
It would be useful to include these tests. Why were they comment out? If there is a pending issue with them, please add pytest markers.

```suggestion
@pytest.mark.xfail
def test_dump_fchk_from_wfn_fci_lih_cation(tmpdir):
    check_load_dump_consistency(tmpdir, 'lih_cation_fci.wfn')


@pytest.mark.xfail
def test_dump_fchk_from_wfn_fci_lif(tmpdir):
    check_load_dump_consistency(tmpdir, 'lif_fci.wfn')


```

Reply:
These two tests, and the other two which are all now marked with `@pytest.mark.xfail`, were commented out because they were converting WFN/WFX file formats containing post-HF wave-functions (and fractional orbital occupations and natural orbitals) to FCHK. So, when dumping FCHK and then loading it to compare to the initial WFN/WFX, the MO occupations do not match, so I commented these tests and opened issue #174 (not knowing that FCHK cannot store natural orbitals). I am not sure how this needs to be fixed, and in the meanwhile we can probably just raise an error for now, until we establish our conventions.